### ISSUE TITLE
Dragged abilities require a left or right click afterward to release

### ISSUE THREAD

**Issue by raysmith59** [brittyazel/Neuron#356]
Currently, when you start dragging an ability, you can release the left mouse button, continue dragging, and then press the left mouse button again to put it on a bar or press the right mouse button to "destroy" the dragged ability.  This is relatively recent behaviour, in the past dragging required holding down the left mouse button and if the mouse button was released while over an appropriate bar then it would be re-positioned there, or if you released the mouse button over nothing it would destroy the dragged ability.

If this behaviour was changed intentionally to avoid some other bug then I don't object to the change but I still haven't quite gotten used to this new behaviour and would prefer to have it changed back if there won't be negative side effects of that change.

**Comment by brittyazel** [comment]
I believe this is usually a side effect of show/hide grid being enabled. I'll have to go through at some point and clean up all of the drag and drop code. That was some of the first code I touched when I took over the project 2 years ago, and I think it's fair to say I've progressed a bit since then lol. Probably worth a re-look.

Also, I cannot state this enough, but I really need more of me. I have a list a mile long of things to fix and only two hands to do the typing :-(

**Comment by DejayRezme** [comment]
I love neuron and think you do awesome work, so please don't get me wrong. But the drag and drop behavior is one of the worst and most infuriating things I've ever experienced with a wow addon. Example:

I drag an ability to empty button. I then change my mind later and drag a new ability to that spot. I go do combat, try to use that ability, it starts switching BACK to tbe previous ability and/or does some weird toggle. I repeat the process again only to find neuron mocking me by repeatedly switching back and forth!

Of course I figured it out by now, you cannot use right click to destroy a dragged ability. If you use left click it works fine. The button remembers that you haven't dismissed the drag and drop properly and keeps putting the ability on your mouse cursor, even when using keybinds.
Not sure why but I think of right click as "cancel" when dragging ability (like moving an item). 
But this behavior drove me crazy for a while because it happens in the worst circumstances and it's not apparent and involves muscle memory of how you drag and drop in wow. And I still get it wrong sometimes for some reason. 

Also if you lock the actionbar with shift it works but that isn't enabled by default (even though it is in wow options).

Again, hanks for the awesome work on neuron, it's brilliant but I feel for beginners this is a potential pitfall.

**Comment by brittyazel** [comment]
Yes, Shadowlands exposed a lot of bugs, some old and some new. This is one that I really want to fix but have about 10,000 other obligations standing in the way. All I can say is that I'm sorry for the inconvenience, and I'm trying. Getting your PhD while also maintaining 3 addons is hard, and it means that I hardly even get to play the game anymore. I'll do my best to keep fixing things, but believe me when I say that any help would be appreciated.

**Comment by DejayRezme** [comment]
Please don't stress over this :) I'm thankful for all the cool addons out there.

I'm going to try to fix some of these issues myself. So I started by making some bug reports of bugs so I can remember.

**Comment by DejayRezme** [comment]
On an added note, I have two more bugs with this. 

1. Using the "Sunn Viewpoint Art" to reduce the size of the worldframe and rendering load, you cannot drop the spell with the left mouse button. You would have to use the "GlobalFXBackgroundModelScene" which seems to be lowest in the "LOW" layer

2. If the bar is locked with shift and you drop a new spell or mount onto an old one, you don't get a satisfying pickup sound. This is not functionally relevant but confusing and makes you wonder if something is broken. I haven't found anything in the code yet that could explain this.
